### DISCUSSION TITLE
What is the license for bootstrap examples? MIT or CC?

### DISCUSSION THREAD

**Discussion by carlosmintfan** [twbs/bootstrap#40609] [Q&A]
Bootstrap code is licensed MIT, docs are Creative Commons Attribution 3.0 Unported License. Why even make that difference as both basically do the same – require attribution – although the MIT license is much shorter and better? But the code from the docs is also CC, not MIT, like the color modes toggle js. But what about the https://getbootstrap.com/docs/5.3/examples/dashboard/dashboard.css, for example? Is it MIT or CC?

**Comment by knarrff** [comment]
Hi @carlosmintfan 

Why two different licenses have been chosen is a question to ask to the authors, but I think in this case it does make sense:

MIT is a license made for software. It does include "and associated documentation" in the license text, but this would still be awkward should someone want to use only the documentation someplace else, as there wouldn't be any software associated with it.

CC licenses, except for CC0, can technically be used for software, but this is explicitly discouraged (see https://creativecommons.org/faq/#Can_I_use_a_Creative_Commons_license_for_software.3F).

So, not knowing which part users might want to re-use (software, docs, possibly both), it is a general recommendation to use a license made for software for that, and a license made for data for documentation. Sometimes what is what is not that easy to distinguish, but that is another matter. In terms of open source, the general recommendation is "a license approved by OSI or FSF" for software and a CC-license for everything else. Which of those depends on your specific needs and wishes.

As to CSS: CSS defines properties of a document. I would in general not count that as software: CSS can't be "executed".

**Reply by carlosmintfan** [reply]
Hi @knarrff 
Yeah, they should change the license of their Color modes toggler and the CSS to MIT.
Stop that precision, of course CSS is software! 😀 
And, in my opinion, docs should be MIT. If somebody wants to reuse the docs, just put the MIT license text and copyright info in the imprint, state that the docs from this and that project have been used, and nothing else. Maybe MIT needs more interpretation than CC when it comes to courts, but...why do we need that! And as the software is also MIT...just go with MIT.
And if I'm right, it's the same with Apache: Software is Apache license, docs are Apache license.
